### ISSUE TITLE
Got different results of the same genome from the different runs of checkm2

### ISSUE THREAD

**Issue by quliping** [chklovski/CheckM2#103]
Hello, I found some of my MAGs have different completeness and contamination in different runs of checkm2. I first get 20 original MAGs and run checkM2 for them. In this run, e.g., the completeness and contamination of MAG bin.SY48.20 are 50.41%/6.91%. Next, I run the reassembly moddule of metawrap and got 20 reassembled version of the 20 original MAGs. I run checkM2 for the 40 MAGs and the completeness and contamination of the original version of bin.SY48.20 became to 49.87%/6.65%. Why the quality score of the same genome changed in the different runs of checkM2 with different input MAG sets? 

this is the first run for the 20 original MAGs:

this is the second run for the 20 original (with 'orig' in the genome id, e.g., 'bin.SY48.20.orig' is the same genome of 'bin.SY48.20' in the first run):

Here is the other run of checkM2 using only two MAGs including the original and reassembled version of  bin.SY48.20, and you can see that the the completeness and contamination were changed again:

**Comment by chklovski** [comment]
Hmm, could potentially be related to https://github.com/keras-team/keras/issues/12400 or the BatchNormalization layers as part of the neural network model, leading to minor fluctuations in output. 

Does this affect the 'General' (Gradient boost) results or only the 'Specific' (Neural network') results for your MAGs? You can test this running the --allmodels command. Are the differences in results in excess of 1% completeness or contamination? 

Will see if I can reproduce it with my own MAGs.

**Comment by quliping** [comment]
> Hmm, could potentially be related to [keras-team/keras#12400](https://github.com/keras-team/keras/issues/12400) or the BatchNormalization layers as part of the neural network model, leading to minor fluctuations in output.
> 
> Does this affect the 'General' (Gradient boost) results or only the 'Specific' (Neural network') results for your MAGs? You can test this running the --allmodels command. Are the differences in results in excess of 1% completeness or contamination?
> 
> Will see if I can reproduce it with my own MAGs.

I run these three MAG sets again with the ` --allmodels` command. Both 'General' and 'Specific' results of bin.SY48.20 were changed in the different runs ('2 MAGs' and '40 MAGs') of checkm2.
![image](https://github.com/chklovski/CheckM2/assets/39583607/e6bb84cf-2d8a-49b0-8bf5-3e0a33b41ea4)
![image](https://github.com/chklovski/CheckM2/assets/39583607/074152b1-3abd-4c96-94e1-8c1c7afc932e)

 However, I don't know why that I could not reproduce the previous results of the '20 original MAGs' (the first run mentioned above)... 
![image](https://github.com/chklovski/CheckM2/assets/39583607/a1ad1fe0-93eb-481b-94cc-e8196864fca8)

**Comment by quliping** [comment]
> Hmm, could potentially be related to [keras-team/keras#12400](https://github.com/keras-team/keras/issues/12400) or the BatchNormalization layers as part of the neural network model, leading to minor fluctuations in output.
> 
> Does this affect the 'General' (Gradient boost) results or only the 'Specific' (Neural network') results for your MAGs? You can test this running the --allmodels command. Are the differences in results in excess of 1% completeness or contamination?
> 
> Will see if I can reproduce it with my own MAGs.

Hello, may I ask if this issue has been resolved? The fluctuation of checkm2 result is no matter for genome with high quality but stilll has influence for genome near the quality threshould. I'm worried that others may come to different conclusions from me when reanalyzing my data. I'm not sure if it's appropriate to publish these genomes. Looking forward for your reply.

**Comment by Zlouisson1** [comment]
I ran checkm2 on my bins and then filtered those <80% completeness and >5% contamination. No other changes expect selecting those high-quality bins. I then ran checkm2 again on the bins that met those specifications, however, there are really big differences in both completeness and contamination scores between the two runs for the same bins. Out of ~900 high quality bins based on the first checkm2 run, about half of them are different. Here are the first 15 bins. 

![Image](https://github.com/user-attachments/assets/5b5918cb-c92a-42a5-a0d5-5e31735ec87a)
